### PR TITLE
[ONNX] Extended ReduceMax by opsets 13,18,20

### LINKED ISSUE #20555: [Good First Issue]: Align behavior of ONNX Frontend operator ReduceMax-11, 12, 13, 18, 20 with original framework
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires alignment between OpenVINO ONNX Frontend and original framework implementations of ReduceMax for next list of opsets: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-11),  [opset 12](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-12),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-18),  [opset 20](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-20)
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#ReduceMax)
More details can be found in ONNX Changelog: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-11),  [opset 12](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-12),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-18),  [opset 20](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceMax-20)
1. Operator already has a common implementation in OpenVINO. First of all, you need to review a documentation and prepare a table with differences between versions. It could be, for instance, a missing property, extended/reduced coverage of existing property, etc...
2. Copy existing implementation [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op) to make it aligned with original framework (extend or reduce coverage of a common implementation). Copy of modified implementation should be in a defined opset, or in opset 1 in case it implements oldest implementation. [Example of multi-opset operation](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/softmax.cpp#L50).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Example Pull Requests

_No response_

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [User documentation](https://docs.openvino.ai/)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor 

### Ticket

_No response_

### PR DESCRIPTION
**Pull request by gkrivor**
### Details:
 - Extended ReduceMax by opsets 13,18,20
 - Updated a using opset for ONNX to 20
 - Added tests for additional supported types
 - Enabled backend tests

### Tickets:
  - Closes #20555

### CODE REVIEW

File: src/frontends/onnx/frontend/src/op/reduce.cpp [hunk 1]
```diff
@@ -94,6 +95,27 @@ const std::set<element::Type> supported_types_v1 =
     {element::u32, element::u64, element::i32, element::i64, element::f16, element::f32, element::f64};
 const std::set<element::Type> supported_types_v2 =
     {element::u32, element::u64, element::i32, element::i64, element::f16, element::f32, element::f64, element::bf16};
+const std::set<element::Type> supported_types_v3 = {element::u32,
```
Comment (mwilczy) [inline src/frontends/onnx/frontend/src/op/reduce.cpp:98]: Hi,
I'm working on new opsets for ReduceLogSumExp, and would like to take this PR as example. 
I'm wondering whether inventing new versioning scheme for types is a good idea ? Wouldn't it be better to just put the constant supported_types into a namespace ?

File: src/frontends/onnx/frontend/src/op/reduce.hpp [hunk 2]
```diff
@@ -29,6 +29,15 @@ ov::OutputVector reduce_l2(const ov::frontend::onnx::Node& node);
 namespace set_1 {
 ov::OutputVector reduce_max(const ov::frontend::onnx::Node& node);
 }  // namespace set_1
+namespace set_13 {
```
Comment (mwilczy) [inline src/frontends/onnx/frontend/src/op/reduce.hpp:32]: This operator is also present in opset version 11. Is there any reason we're skipping version 11 ? In my case for ReduceLogSumExp there is no change between version 1 and 11, however maybe it would be good to include it for compatibility reasons ?

Comment (gkrivor) [reply src/frontends/onnx/frontend/src/op/reduce.hpp:32]: Hi @mwilczy , I put a common motivation to do not introduce opset-11 for all Reduce* here https://github.com/openvinotoolkit/openvino/blob/c208a881cb4e1889d8d7126b838b4fcb86b44b8d/src/frontends/onnx/frontend/src/op/reduce.cpp#L167

File: src/frontends/onnx/frontend/src/op/reduce.cpp (hunk 1 above)
Comment (gkrivor) [reply src/frontends/onnx/frontend/src/op/reduce.cpp:98]: @mwilczy you mean why I didn't align it with an opset version? For example, supported_types_v1, v11, v13, etc. Or op::set_1::supported_types}, op::set_11::supported_types, etc. Did I get correct? I thought about it at first step but...

I motivated to do it as I did because I compared all differences as a table, and I found by text comparison between all versions of all Reduce* operations. A summary:

Types versioning isn't aligned between Reduce* versions. For example, ReduceMax/Min-13 has different list of supported types (added int8 and uint8) than all other Reduce*-13 operations. as I can see - opset-13 simultaneously introduced two different lists of supported types. I really don't like constructions like op::set_13::supported_types::v1 ;)

So, supported_types for ReduceMax/Min will look different. And in such case I may need to introduce more intermediate entities like: define four different lists in a common part and use it thru... "using support_types = supported_types_v3" for each Reduce*? I think it will overcomplicate the source without any benefits for reading. In my opinion it could be reasonable in case we would separate Reduce* operations for a set of files (maybe it will be done when we will have all implementations ready).

Hope, I've answered on your question...
